library/leon/io/StdIn.scala

```diff
@@ -16,38 +16,38 @@ object StdIn {
   @isabelle.noBody()
   def readInt(implicit state: State): Int = {
     state.seed += 1
-    nativeReadInt
+    nativeReadInt(state.seed)
   }
 
   @library
   @extern
-  private def nativeReadInt(implicit state: State): Int = {
+  private def nativeReadInt(seed: BigInt): Int = {
```

Comment:
Is there a limitation or issue to have a `State` as parameter instead of a `BigInt`? I'm asking this because this would break a lot of stuff for GenC -- currently I'm "dropping" `State` by replacing them with `void*` because they are never used, but I can't do the same with `BigInt`.